src/udev/net/ethtool-util.c

```diff
@@ -75,6 +75,26 @@ static const char* const netdev_feature_table[_NET_DEV_FEAT_MAX] = {
         [NET_DEV_FEAT_UFO]  = "tx-udp-fragmentation",
 };
 
+static const char* const advertise_table[_NET_DEV_ADVERTISE_MAX] = {
+        [NET_DEV_ADVERTISE_10BASET_HALF]        = "10baset-half",
+        [NET_DEV_ADVERTISE_10BASET_FULL]        = "10baset-full",
+        [NET_DEV_ADVERTISE_100BASET_HALF]       = "100baset-half",
+        [NET_DEV_ADVERTISE_100BASET_FULL]       = "100baset-full",
+        [NET_DEV_ADVERTISE_1000BASET_HALF]      = "1000baset-half",
+        [NET_DEV_ADVERTISE_1000BASET_FULL]      = "1000baset-full",
+        [NET_DEV_ADVERTISE_10000BASET_FULL]     = "10000baset-full",
+        [NET_DEV_ADVERTISE_2500BASEX_FULL]      = "2500basex-full",
+        [NET_DEV_ADVERTISE_1000BASEKX_FULL]     = "1000basekx-full",
+        [NET_DEV_ADVERTISE_10000BASEKX4_FULL]   = "10000basekx4-full",
+        [NET_DEV_ADVERTISE_10000BASEKR_FULL]    = "10000basekr-full",
+        [NET_DEV_ADVERTISE_10000BASER_FEC]      = "10000baser-fec",
+        [NET_DEV_ADVERTISE_20000baseMLD2_Full]  = "20000basemld2-full",
+        [NET_DEV_ADVERTISE_20000baseKR2_Full]   = "20000basekr2-full",
+};
+
+DEFINE_STRING_TABLE_LOOKUP(advertise, NetDevAdvertise);
+DEFINE_CONFIG_PARSE_ENUM(config_parse_advertise, advertise, NetDevAdvertise, "Failed to parse advertise setting");
```

Comment:
Should take multiple values. Because NetDevAdvertise is a bitfield.

Reply:
Please elaborate.

Reply:
I think we should accept e.g., `1000` or `1000baset`, if it is possible. If `1000` is provided, then the parsing function sets all flags speed is 1000.

Reply:
Well 1000 could  be 
```  [NET_DEV_ADVERTISE_1000BASET_HALF]      = "1000baset-half",
        [NET_DEV_ADVERTISE_1000BASET_FULL]      = "1000baset-full",


```

Reply:
hmm should we do with extract_work* and do nit wise '|'.